blocks/blocks.php

```diff
@@ -5,35 +5,14 @@
 
 namespace WPCOMVIP\Decoupled\Blocks;
 
-use WPGraphQL;
-
-function preg_match_html_block( $block ) {
-	// Strip wrapping tags from the content and set as a property on the block.
-	// This allows the front-end implementor to delegate tag creation to a
-	// component.
-
-	preg_match( '#^<([A-z][A-z0-9]*)\b([^>])*>(.*?)</\1>$#', $block, $matches );
-	
-	if ( isset( $matches[1] ) ) {
-		return [
-			'inner_html' => $matches[3],
-			'tag_name'   => $matches[1],
-		];
-	}
-
-	// Self closing HTML block
-	preg_match( '#^<([A-z][A-z0-9]*)+?\b(.*?)\/>$#', $block, $self_closing_matches );
-
-	if ( isset( $self_closing_matches[1] ) ) {
-		return [
-			'inner_html' => null,
-			'tag_name'   => $self_closing_matches[ 1 ],
-		];
-	}
-}	
-
-function parse_blocks( $post_model ) {
-	$version = '0.1.0';
+/**
+ * Extract the content blocks and associated meta for a post.
+ *
+ * @param  WPGraphQL\Model\Post $post_model Post model for post.
+ * @return array
+ */
+function get_content_blocks( $post_model ) {
+	$version = '0.2.0';
```

Comment:
I think this is a good way of versioning the API-response and improving communication between developers on both spectrums (WP and Node).

But I believe we don't have a way to make two versions co-exist. Breaking changes to GraphQL APIs will always be so frustrating 😭

Reply:
This change is purely additive (it only adds `innerBlocks` and doesn't modify or remove any existing fields), so luckily this isn't a breaking change since it won't break any existing queries.

For future breaking changes, I agree we need a strategy. WPGraphQL (or GraphQL for that matter) doesn't have any mechanisms for navigating breaking schema changes. WPGraphQL more or less relies on semver and a CHANGELOG to communicate changes.

Generally speaking, I think a good strategy is:

- When adding new types or modifying existing types, always introduce new fields.
- Don't remove or change the type of existing fields. You can [mark them as `@deprecated`](https://www.apollographql.com/docs/apollo-server/schema/directives/) so that they can span the release window, and later remove them after you're sure consumers have migrated to the new fields.

Interestingly, I think we do have the opportunity to help with upstream breaking changes with WPGraphQL and, by extension, `graphql-php`. Since we conditionally include WPGraphQL in code, we could ship multiple versions of the plugin and decide at runtime which one to load. In theory, we could inspect the request for a special query parameter or header and use it to determine which plugin version to load (`X-WPGraphQL-Version: 2.2.0`).